lib/resource-mapper.js

```diff
@@ -68,6 +68,12 @@ class ResourceMapper {
     if (path.startsWith(this._rootPath)) {
       path = path.substring(this._rootPath.length)
     }
+    if (this._includeHost) {
+      if (!path.startsWith(`/${hostname}`)) {
```

Comment:
Needs to start with `/${hostname}/` (otherwise, hostname prefixes would also be allowed).

Reply:
Might want to add a negative test for the slashless case.